### ISSUE TITLE
com.hazelcast.jet.pipeline.SourceBuilderTest.test_faultTolerance_restartTwice[1: mode=client]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast-jet#1454]
_master_  (commit 69cd63dcf7f1f954533954422a526eed1145a332)

Failed on Sonar build: http://jenkins.hazelcast.com/view/Jet/job/jet-oss-master-sonar-nightly/157/testReport/com.hazelcast.jet.pipeline/SourceBuilderTest/test_faultTolerance_restartTwice_1__mode_client_/

Stacktrace:
```
java.lang.AssertionError: jobId=39f2-40a6-bdc2-2a16 expected:<RUNNING> but was:<COMPLETING>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at com.hazelcast.jet.core.JetTestSupport.lambda$assertJobStatusEventually$2(JetTestSupport.java:153)
	at com.hazelcast.jet.core.JetTestSupport$1.run(JetTestSupport.java:196)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1310)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1327)
	at com.hazelcast.jet.core.JetTestSupport.assertTrueEventually(JetTestSupport.java:161)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:152)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:147)
	at com.hazelcast.jet.pipeline.SourceBuilderTest.test_faultTolerance_restartTwice(SourceBuilderTest.java:416)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'test_faultTolerance_restartTwice[1: mode=client](com.hazelcast.jet.pipeline.SourceBuilderTest):'
13:08:10, accumulated pauses: 980 ms, max pause: 578 ms, pauses over 1000 ms: 0
13:08:15, accumulated pauses: 107 ms, max pause: 49 ms, pauses over 1000 ms: 0
13:08:20, accumulated pauses: 31 ms, max pause: 1 ms, pauses over 1000 ms: 0
13:08:25, accumulated pauses: 86 ms, max pause: 5 ms, pauses over 1000 ms: 0
13:08:30, accumulated pauses: 54 ms, max pause: 3 ms, pauses over 1000 ms: 0
13:08:35, accumulated pauses: 237 ms, max pause: 157 ms, pauses over 1000 ms: 0
13:08:40, accumulated pauses: 133 ms, max pause: 16 ms, pauses over 1000 ms: 0
13:08:45, accumulated pauses: 44 ms, max pause: 4 ms, pauses over 1000 ms: 0
13:08:50, accumulated pauses: 114 ms, max pause: 10 ms, pauses over 1000 ms: 0
13:08:55, accumulated pauses: 30 ms, max pause: 2 ms, pauses over 1000 ms: 0
13:09:00, accumulated pauses: 45 ms, max pause: 10 ms, pauses over 1000 ms: 0
13:09:05, accumulated pauses: 52 ms, max pause: 5 ms, pauses over 1000 ms: 0
13:09:10, accumulated pauses: 45 ms, max pause: 13 ms, pauses over 1000 ms: 0
13:09:15, accumulated pauses: 66 ms, max pause: 12 ms, pauses over 1000 ms: 0
13:09:20, accumulated pauses: 96 ms, max pause: 12 ms, pauses over 1000 ms: 0
13:09:25, accumulated pauses: 83 ms, max pause: 12 ms, pauses over 1000 ms: 0
13:09:30, accumulated pauses: 151 ms, max pause: 10 ms, pauses over 1000 ms: 0
13:09:35, accumulated pauses: 89 ms, max pause: 9 ms, pauses over 1000 ms: 0
13:09:40, accumulated pauses: 155 ms, max pause: 19 ms, pauses over 1000 ms: 0
13:09:45, accumulated pauses: 132 ms, max pause: 11 ms, pauses over 1000 ms: 0
13:09:50, accumulated pauses: 70 ms, max pause: 5 ms, pauses over 1000 ms: 0
13:09:55, accumulated pauses: 117 ms, max pause: 14 ms, pauses over 1000 ms: 0
13:10:00, accumulated pauses: 149 ms, max pause: 10 ms, pauses over 1000 ms: 0
13:10:05, accumulated pauses: 120 ms, max pause: 11 ms, pauses over 1000 ms: 0
13:10:10, accumulated pauses: 83 ms, max pause: 7 ms, pauses over 1000 ms: 0
13:10:15, accumulated pauses: 235 ms, max pause: 184 ms, pauses over 1000 ms: 0
```

**Comment by viliam-durina** [comment]
The test is parallel therefore the logs are unusable. Problem doesn't reproduce locally. I glanced over the entire test output, didn't spot anything suspicious. There are two jobs repeatedly doing snapshots at the end, but this is unrelated as all tests in the class share single cluster and some test doesn't cancel the job so it lingers doing snapshots (I confirmed this locally by adding a test that only contains `sleepSeconds(600)` and observed the snapshots after all other tests succeeded).

I suggest to rewrite this test to serial and ensure all jobs are cancelled after it terminates.
